src/views/AddItem.jsx

```diff
@@ -1,3 +1,93 @@
-export function AddItem() {
-	return <p>Hello from the <code>/add-item</code> page!</p>
+import { addItem } from '../api/firebase';
+import { useState } from 'react';
+
+export function AddItem({ listId }) {
+	// declare itemName, daysUntilNextPurchase, and radioSelect
+	// into separate state variables
+	const [itemName, setItemName] = useState('');
+
+	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(null);
+
+	const [radioSelect, setRadioSelect] = useState('soon');
+
+	// Store values inside itemData and sent over to database
+	// otherwise log an error if request fails
+	const onFormSubmit = (event) => {
+		event.preventDefault();
+		const itemData = {
+			itemName,
+			daysUntilNextPurchase,
+		};
+		try {
+			addItem(listId, itemData);
+			alert(`${itemData.itemName} was added to your list`);
+		} catch (error) {
+			console.log(error);
+		}
+	};
+
+	// Get user input and set itemName
+	const handleItemName = (event) => {
+		setItemName(event.target.value);
+	};
+
+	// Get radio selection and set daysUntilNextPurchase
+	const handlePurchaseDate = (event) => {
+		let date = null;
+		if (event.target.value === 'soon') {
+			date = 7;
+		} else if (event.target.value === 'kind-of-soon') {
+			date = 14;
+		} else if (event.target.value === 'not-soon') {
+			date = 30;
+		}
+		setDaysUntilNextPurchase(date);
+		setRadioSelect(event.target.value);
+	};
+
+	return (
+		<form onSubmit={onFormSubmit}>
+			<div>
+				<label htmlFor="item">Item Name:</label>
+			</div>
+			<input type="text" id="item" name="item" onChange={handleItemName} />
```

Comment:
For `onChange={handleItemName}`, another way to write this is by using `onChange={(e) => setItemName(e.target.value)}` which would allow you to remove the `handleItemName` function. I'm not sure if this is best practice, however, and it could lessen readability.

```suggestion
			<input type="text" id="item" name="item" onChange={(e) => setItemName(e.target.value)} />
```

Reply:
@alex-andria Thanks! Great suggestion and will implement!